mateog-assign1/crypto.py

```diff
@@ -0,0 +1,322 @@
+            k = ord(keyword[i % len(keyword)]) - ord('A')
+            ciphertext[i] = chr(((p + k) % 26) + ord('A'))
+
+        return "".join(ciphertext)
+
+
+def decrypt_vigenere(ciphertext, keyword):
+        """
+        Decrypts ciphertext using a Vigenere cipher with a keyword. Creates
+        plaintext list of same length as ciphertext, gets ASCII value of
+        each letter in ciphertext and corresponding letter in keyword, subtracts
+        value of 'A' from each, subtracts key character value from ciphertext value,
+        mods sum by 26 to get a wrapping shift, and adds 'A' back to sum before
+        getting ASCII character back from plaintext value.
+        """
+        ciphertext = ciphertext.upper()
+        keyword = keyword.upper()
+        plaintext = list(ciphertext)
+        for i in range(0, len(ciphertext)):
+            p = ord(ciphertext[i]) - ord('A')
+            k = ord(keyword[i % len(keyword)]) - ord('A')
+            plaintext[i] = chr(((p - k) % 26) + ord('A'))
+
+        return "".join(plaintext)
+
+
+def encrypt_railfence(plaintext, num_rails):
+        """
+        Encrypts plaintext using a railfence cipher. Sets plaintext to uppercase,
+        and loops through each character in plaintext to determine its rail. First
+        splits characters into fragments, which include the character at a peak/trough
+        (inclusive) through the character at the next peak/trough (exclusive). Next,
+        determines whether fragment is descending or ascending, and splits characters
+        from the start of the fragment to the end into their corresponding rail of
+        increasing index, or the opposite, respectively.
+        """
+        plaintext = plaintext.upper()
+        frag_len = num_rails - 1
+        rails = [""] * num_rails
+        for i in range(0, len(plaintext)):
+            frag_num = int(i / frag_len)
+            if frag_num % 2 == 0:
+                rails[i % frag_len] += plaintext[i]
+            else:
+                rails[-(1 + i % frag_len)] += plaintext[i]
+
+        ciphertext = ""
+        for r in rails:
+            ciphertext += r
+
+        return ciphertext
+
+
+def decrypt_railfence(ciphertext, num_rails):
+        """
+        Encrypts plaintext using a railfence cipher.
+        Add more implementation details here.
+        """
+        ciphertext = ciphertext.upper()
+        frag_len = num_rails - 1
+        rails = [""] * num_rails
+        num_full_frags = int(len(ciphertext) / frag_len)
+        len_last_frag = len(ciphertext) % frag_len
+        for i in range(0, num_rails):
+
+            # TOP RAIL
+            if i == 0:
+                num_peaks = int(num_full_frags / 2)
+                if num_full_frags % 2 != 0 or len_last_frag > 0: # If extra letter from full/part fragment
+                    num_peaks += 1
+                rails[i] = ['-'] * num_peaks
+
+            # BOTTOM RAIL
+            elif i == num_rails - 1:
```

Comment:
This case seems to do the same thing as the above - would suggest combining the cases together